cmd/gapic-showcase/compliance_suite_test.go

```diff
@@ -0,0 +1,201 @@
+					continue
+				}
+
+				// Check for expected response.
+				if got, want := response.GetInfo(), requestProto.GetInfo(); !proto.Equal(got, want) {
+					gotString := jsonMarshaller.Format(got)
+					wantString := jsonMarshaller.Format(want)
+					t.Errorf("%s unexpected response:\n   -->got:\n`%s`\n   -->want:\n`%s`\n",
+						errorPrefix, gotString, wantString)
+				}
+			}
+		}
+	}
+}
+
+// The following are helpers for TestComplianceSuite, since Showcase doesn't intrinsically define a
+// REST client. Each GAPIC generator should instead use the GAPIC it generated for the Showcase
+// API.
+type prepRepeatDataTestFunc func(request *genproto.RepeatRequest) (verb string, name string, path string, body string, err error)
+
+func prepRepeatDataBodyTest(request *genproto.RepeatRequest) (verb string, name string, path string, body string, err error) {
+	name = "Compliance.RepeatDataBody"
+	bodyBytes, err := protojson.Marshal(request)
+	return name, "POST", "/v1beta1/repeat:body", string(bodyBytes), err
+}
+
+func prepRepeatDataQueryTest(request *genproto.RepeatRequest) (verb string, name string, path string, body string, err error) {
+	name = "Compliance.RepeatDataQuery"
+	queryString := prepRepeatDataTestsQueryString(request, nil)
+	return name, "GET", "/v1beta1/repeat:query" + queryString, body, err
+}
+
+func prepRepeatDataSimplePathTest(request *genproto.RepeatRequest) (verb string, name string, path string, body string, err error) {
+	name = "Compliance.RepeatDataSimplePath"
+	info := request.GetInfo()
+
+	// TODO: Determine behavior for a string field path param whose value is empty. This should be
+	// a failure, probably, in which case we need to augment the ComplianceGroup to allow
+	// specifying expected errors.
+	// TODO: Add to compliance_suite cases with near-maximal values.
+	path = fmt.Sprintf("/v1beta1/repeat/%s/%d/%.20g/%t:simplepath",
+		url.PathEscape(info.GetFString()), info.GetFInt32(), info.GetFDouble(), info.GetFBool())
+
+	// exclude the path fields from the query params
+	exclude := map[string]bool{
+		"f_string": true,
+		"f_int32":  true,
+		"f_double": true,
+		"f_bool":   true,
+	}
+	queryString := prepRepeatDataTestsQueryString(request, exclude)
+	return name, "GET", path + queryString, body, err
+}
+
+// prepRepeatDataTestsQueryString returns the query string containing all fields in `request.info`
+// except for those whose proto name (relative to request.info) are present in the `exclude` map
+// with a value of `true`.
+func prepRepeatDataTestsQueryString(request *genproto.RepeatRequest, exclude map[string]bool) string {
+	info := request.GetInfo()
+	queryParams := []string{}
+	addParam := func(condition bool, key, value string) {
+		if !condition {
+			return
+		}
+		queryParams = append(queryParams, fmt.Sprintf("info.%s=%s", key, value))
+	}
+
+	if !exclude["f_string"] {
+		addParam(len(info.GetFString()) > 0, "f_string", strings.ReplaceAll(strings.ReplaceAll(info.FString, " ", "+"), "%", "%%"))
```

Comment:
Nit: There is one more `info.FString` to replace in this line, at the end. Sorry! GitHub comments don't allow me to highlight sections of a line.

Reply:
Ah, thanks!